consensus/src/chained_bft/block_storage/sync_manager.rs

```diff
@@ -187,7 +187,14 @@ impl<T: Payload> BlockStore<T> {
         );
         let mut quorum_certs = vec![];
         quorum_certs.push(highest_commit_cert.clone());
```

Comment:
what if instead we just took line 190 and deleted 189 and 190-197? and we could add an assert that qc[0] (I think) is the same as highest_commit_cert?

Reply:
let me assert the id then

Reply:
wait, I'm confused... your approach match my suggestion either lol.... also you eliminated the assertion that block.len == qc.len
now we're just nitpicking on the code... so I'll accept but I would think... the best solution is....

let quorum_certs = blocks.iter().take(3).map(|b| b.quorum_cert().clone());
assert_eq!(quorum_cert[0], highest_commit_cert);

then you delete a whole lot of code...

Reply:
your code is not equivalent to the current code
imagine the blocks [C0 <- B1] <- [C1 <- B2] <- [C2 <- B3] <- C3
blocks is [B1, B2, B3],
what we want is [C1, C2, C3], what blocks.iter().take(3).map(|b| b.quorum_cert().clone()) gets is [C0, C1, C2]